src/System.Net.Http/tests/FunctionalTests/DefaultCredentialsTest.cs

```diff
@@ -0,0 +1,170 @@
+            using (HttpResponseMessage response = await client.GetAsync(AuthenticatedServer))
+            {
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                
+                string responseBody = await response.Content.ReadAsStringAsync();
+                VerifyAuthentication(responseBody, true, SpecificDomain + "\\" + SpecificUserName);
+            }
+        }
+
+        [ConditionalTheory(nameof(HttpInternalTestsEnabled))]
+        // TODO: Issue #8176 [InlineData(false)]
+        [InlineData(true)]
+        public async Task Credentials_SetToWrappedDefaultCredential_ConnectAsCurrentIdentity(bool useProxy)
+        {
+            var handler = new HttpClientHandler();
+            handler.UseProxy = useProxy;
+            handler.Credentials = new CredentialWrapper
+            {
+                InnerCredentials = CredentialCache.DefaultCredentials
+            };
+
+            using (var client = new HttpClient(handler))
+            using (HttpResponseMessage response = await client.GetAsync(AuthenticatedServer))
+            {
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+                string responseBody = await response.Content.ReadAsStringAsync();
+                WindowsIdentity currentIdentity = WindowsIdentity.GetCurrent();
+                _output.WriteLine("currentIdentity={0}", currentIdentity.Name);
+                VerifyAuthentication(responseBody, true, currentIdentity.Name);
+            }
+        }
+
+        private void VerifyAuthentication(string response, bool authenticated, string user)
+        {
+            // Convert all strings to lowercase to compare. Windows treats domain and username as case-insensitive.
+            response = response.ToLower();
+            user = user.ToLower();
+            _output.WriteLine(response);
+
+            if (!authenticated)
+            {
+                Assert.True(
+                    TestHelper.JsonMessageContainsKeyValue(response, "authenticated", "false"),
+                    "authenticated == false");
+            }
+            else
+            {
+                Assert.True(
+                    TestHelper.JsonMessageContainsKeyValue(response, "authenticated", "true"),
+                    "authenticated == true");
+                Assert.True(
+                    TestHelper.JsonMessageContainsKeyValue(response, "user", user),
+                    $"user == {user}");
+            }
+        }
+
+        private class CredentialWrapper : ICredentials
+        {
+            public ICredentials InnerCredentials { get; set; }
+
+            public NetworkCredential GetCredential(Uri uri, String authType)
+            {
+                if (InnerCredentials != null)
+                {
+                    return InnerCredentials.GetCredential(uri, authType);
+                }
+                else
+                {
+                    return null;
+                }
```

Comment:
I originally just cut/pasted the repro code from the original bug report. But I'll simplify as you suggest.